src/loaders/genome_collection/parse_biolog_data.py

```diff
@@ -0,0 +1,187 @@
+    meta_df = _read_excel_as_df(biolog_meta_file)
+
+    return meta_df
+
+
+def _retrieve_kbase_genome_id(meta_df, strain_id) -> str:
+    # Retrieve the KBase genome ID for the given strain ID from the metadata file (dataframe).
+
+    matching_row = meta_df[(meta_df['Strain ID'] == strain_id) & meta_df['Genome_Ref'].notna()]
+
+    if matching_row.empty:
+        raise ValueError(f"No matching Strain ID with a non-NaN Genome_Ref found for {strain_id}.")
+    else:
+        genome_ref = matching_row['Genome_Ref'].values[0]
+
+        return genome_ref
+
+
+def generate_biolog_heatmap_data(
+        biolog_data_file: Path,
+        biolog_meta_file: Path,
+        load_ver: str,
+        env: str = 'PROD',
+        kbase_collection: str = 'biolog',
+        root_dir: str = loader_common_names.ROOT_DIR):
+    """
+    Generate JSONL files with the heatmap data for the Biolog data file.
+
+    :param biolog_data_file: Biolog data file
+    :param biolog_meta_file: Biolog meta file
+    :param load_ver: the version of the data to be loaded
+    :param env: the environment containing the data to be processed
+    :param kbase_collection: the name of the KBase collection
+    :param root_dir: the root directory for the data to be processed
+    """
+    data_df = _read_biolog_data(biolog_data_file)
+    meta_df = _read_biolog_meta(biolog_meta_file)
+
+    heatmap_cell_details, heatmap_rows, heatmap_meta_dict = list(), list(), dict()
+
+    # create document for heatmap meta
+    categories = list()
+    for idx, metabolite in enumerate(data_df.columns):
+        # making each metabolite a category
+        categories.append({FIELD_HEATMAP_CATEGORY: metabolite,
+                           FIELD_HEATMAP_COLUMNS: [{FIELD_HEATMAP_COL_ID: str(idx),
+                                                    FIELD_HEATMAP_NAME: metabolite,
+                                                    FIELD_HEATMAP_DESCR: metabolite,
+                                                    FIELD_HEATMAP_TYPE: ColumnType.BOOL.value
+                                                    }]
+                           })
+    heatmap_meta_dict[FIELD_HEATMAP_CATEGORIES] = categories
+    heatmap_meta_dict[FLD_COLLECTION_ID] = kbase_collection
+    heatmap_meta_dict[FLD_LOAD_VERSION] = load_ver
+    heatmap_meta_dict[FLD_ARANGO_KEY] = collection_load_version_key(kbase_collection, load_ver)
+
+    min_value, max_value = float('inf'), float('-inf')
+    for strain_id, row in data_df.iterrows():
+        try:
+            genome_ref = _retrieve_kbase_genome_id(meta_df, strain_id)
+        except ValueError as e:
+            # TODO: this is a temporary solution to handle the missing KBase genome ID for some strains.
+            # Two strains 'PDO1076' and 'PTD-1' are missing the KBase genome ID in the metadata file.
+            print(f'Cannot find KBase genome ID for strain ID: {strain_id}.')
+            genome_ref = f'kbase_id_not_found - {str(uuid.uuid4())}'
```

Comment:
For now I would just skip them. The server should handle that ok, it expects that not every row in genome attribs will have heatmap data

Reply:
👍

Reply:
:+1: